### PR TITLE
USHIFT-3780: Reorganized test harness scripts

### PR DESCRIPTION
**Pull request by vanhalenar**

### CODE REVIEW

File: test/bin/ci_phase_iso_build.sh [hunk 1]
```diff
@@ -116,7 +116,6 @@ cd "${ROOTDIR}"
 # Get firewalld and repos in place. Use scripts to get the right repos
 # for each branch.
 $(dry_run) bash -x ./scripts/devenv-builder/configure-vm.sh --no-build --force-firewall "${PULL_SECRET}"
-$(dry_run) bash -x ./scripts/image-builder/configure.sh
```
Comment (pmtk) [inline test/bin/ci_phase_iso_build.sh:119]: Problem with this is that `build_rpms.sh` uses things installed in this script, like `parallel`. We could move whole `if ${COMPOSER_CLI_BUILDS}` to this place,  but we also need to install some things for bootc, so starting osbuild workers is a bit pointless, but if we want to keep the script `manage_composer_config.sh create` then we need to do it, I guess

Comment (pmtk) [reply test/bin/ci_phase_iso_build.sh:119]: I think an alternative would be to have separate args:
```
manage_composer_config.sh create
manage_composer_config.sh start-osbuild-workers N
```
So we always execute first one, and we only execute second one `if ${COMPOSER_CLI_BUILDS}`

Comment (vanhalenar) [reply test/bin/ci_phase_iso_build.sh:119]: I added a command `manage_composer_config.sh create-workers <num>` to just create the osbuild workers.

File: test/bin/manage_composer_config.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    <num_workers>: Number of workers.
+
+EOF
+}
+
+action_create() {
+    "${ROOTDIR}/scripts/image-builder/configure.sh"
+
+    # Optionally create workers for building in parallel
+    if [ $# -ne 0 ]; then
```
Comment (ggiguash) [inline test/bin/manage_composer_config.sh:36]: Let's add a check that we actually have `create-workers` command line.

File: test/bin/start_webserver.sh [hunk 3]
```diff
@@ -9,6 +9,36 @@ SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 # shellcheck source=test/bin/common.sh
 source "${SCRIPTDIR}/common.sh"
 
+usage() {
+    cat - <<EOF
+${BASH_SOURCE[0]} [stop]
```
Comment (ggiguash) [inline test/bin/start_webserver.sh:14]: Nit: for consistency, let's rename this script to manage_webserver.sh? 
Also, we need `[start | stop]` or ``[create | cleanup]` arguments for consistency with other cleanup scripts.

File: test/bin/ci_phase_iso_build.sh [hunk 4]
```diff
@@ -130,16 +130,13 @@ $(dry_run) bash -x ./bin/build_rpms.sh
...
-$(dry_run) bash -x ./bin/start_webserver.sh
-
 if ${COMPOSER_CLI_BUILDS} ; then
     # Figure out an optimal number of osbuild workers
     CPU_CORES="$(grep -c ^processor /proc/cpuinfo)"
     MAX_WORKERS=$(find "${ROOTDIR}/test/image-blueprints" -name \*.toml | wc -l)
     CUR_WORKERS="$( [ "${CPU_CORES}" -lt  $(( MAX_WORKERS * 2 )) ] && echo $(( CPU_CORES / 2 )) || echo "${MAX_WORKERS}" )"
 
-    $(dry_run) bash -x ./bin/start_osbuild_workers.sh "${CUR_WORKERS}"
+    $(dry_run) bash -x ./bin/manage_composer_config.sh create-workers "${CUR_WORKERS}"
```
Comment (ggiguash) [inline test/bin/ci_phase_iso_build.sh:139]: Let's move the core math to the `manage_composer_config.sh` script and get rid of the `<num>` argument?

File: test/bin/ci_phase_iso_build.sh (hunk 1 above)
Comment (ggiguash) [reply test/bin/ci_phase_iso_build.sh:119]: It's a good approach. Small nit: let's get rid of the `<num> arg as I commented below?

File: test/bin/manage_composer_config.sh (hunk 2 above)
Comment (ggiguash) [inline test/bin/manage_composer_config.sh:36]: Let's remove this implicit configuration as we now have `create-workers` option?

File: test/bin/manage_hypervisor_config.sh [hunk 5]
```diff
@@ -10,16 +10,22 @@ source "${SCRIPTDIR}/common.sh"
...
         Uses the VM_POOL_BASENAME, VM_DISK_BASEDIR and
         VM_ISOLATED_NETWORK variables.
 
-cleanup: Undo the settings made by 'create' command.
+cleanup: Undo the settings made by 'create' command, 
+         stop the nginx file-server.
 
+cleanup-all: Clean up all scenario infrastructure, 
+             revert the hypervisor configuration and 
+             kill the web server process.
```
Comment (ggiguash) [inline test/bin/manage_hypervisor_config.sh:28]:
```suggestion
cleanup-all: Clean up all scenario infrastructure
             and undo the settings made by 'create' command
```

File: test/bin/manage_hypervisor_config.sh [hunk 6]
```diff
@@ -10,16 +10,22 @@ source "${SCRIPTDIR}/common.sh"
...
-create: Set up firewall, storage pool and network.
+create: Set up firewall, storage pool and network. 
+        Start nginx file-server to serve images 
+        for test scenarios.
         Uses the VM_POOL_BASENAME, VM_DISK_BASEDIR and
         VM_ISOLATED_NETWORK variables.
 
-cleanup: Undo the settings made by 'create' command.
+cleanup: Undo the settings made by 'create' command, 
+         stop the nginx file-server.
```
Comment (ggiguash) [inline test/bin/manage_hypervisor_config.sh:24]:
```suggestion
cleanup: Undo the settings made by 'create' command.
```

**Comment by ggiguash** [comment]
Let's also remote the unnecessary scripts in this PR:
- start_osbuild_workers.sh
- rebuild_source_images.sh
- cleanup_composer.sh
- cleanup_hypervisor.sh

File: test/bin/ci_phase_iso_build.sh (hunk 4 above)
Comment (vanhalenar) [reply test/bin/ci_phase_iso_build.sh]: Done, I made it so when running `manage_composer_config.sh create-workers`, it determines the ideal number of workers, but I left the option to create a specific number of workers by running `manage_composer_config.sh create-workers [number]`.

File: test/bin/manage_composer_config.sh (hunk 2 above)
Comment (vanhalenar) [reply test/bin/manage_composer_config.sh:34]: Removed

File: test/bin/manage_hypervisor_config.sh (hunk 5 above)
Comment (vanhalenar) [reply test/bin/manage_hypervisor_config.sh]: Changed

File: test/bin/manage_hypervisor_config.sh (hunk 6 above)
Comment (vanhalenar) [reply test/bin/manage_hypervisor_config.sh]: Changed

File: test/bin/manage_composer_config.sh [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    [num_workers]: Number of workers. If unspecified,
+                   The optimal number of workers will
+                   be determined based on the number
+                   of CPU cores.
+
+EOF
+}
+
+action_create() {
+    if [ $# -ne 0 ]; then
```
Comment (ggiguash) [inline test/bin/manage_composer_config.sh:34]: Let's move the argument check out of the function where args are parsed.

File: test/bin/manage_composer_config.sh [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    
+    "${TESTDIR}/bin/manage_webserver.sh" "start"
+}
+
+action_create-workers() {
+    # If no number is given, determine the optimal number of workers
+    if [ $# -eq 0 ]; then
+        CPU_CORES="$(grep -c ^processor /proc/cpuinfo)"
+        MAX_WORKERS=$(find "${ROOTDIR}/test/image-blueprints" -name \*.toml | wc -l)
+        CUR_WORKERS="$( [ "${CPU_CORES}" -lt  $(( MAX_WORKERS * 2 )) ] && echo $(( CPU_CORES / 2 )) || echo "${MAX_WORKERS}" )"
```
Comment (ggiguash) [inline test/bin/manage_composer_config.sh:49]: Please, convert these var names to lower case and use `local -r var_name` declaration

File: test/bin/manage_webserver.sh [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+action_start() {
+    echo "Starting web server in ${IMAGEDIR}"
+    mkdir -p "${IMAGEDIR}"
+    cd "${IMAGEDIR}"
+
+    NGINX_CONFIG="${IMAGEDIR}/nginx.conf"
+    # See the https://nginx.org/en/docs/http/ngx_http_core_module.html page for
+    # a full list of HTTP configuration directives
+    echo "${nginx_conf}" > "${NGINX_CONFIG}"
```
Comment (ggiguash) [inline test/bin/manage_webserver.sh:69]: If you're using echo to create a file, I think all the formatting will go away.
Let's use in-place editing?

Comment (ggiguash) [reply test/bin/manage_webserver.sh:69]: You can separate it into a function if you'd like to simplify the current function code.

Comment (vanhalenar) [reply test/bin/manage_webserver.sh:69]: I think it keeps the formatting, as long as you're using double quotes around the variable: `"${nginx_conf}"`

File: test/bin/manage_composer_config.sh (hunk 8 above)
Comment (vanhalenar) [reply test/bin/manage_composer_config.sh]: Done

File: test/bin/manage_composer_config.sh [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+        workers="${CUR_WORKERS}"
```
[the remark this answers is not in the document]
Comment (vanhalenar) [reply test/bin/manage_composer_config.sh]: Done

File: test/bin/manage_webserver.sh (hunk 9 above)
Comment (vanhalenar) [reply test/bin/manage_webserver.sh]: Changed to be more consistent

File: test/bin/manage_webserver.sh [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+    # Default: 60s
+    send_timeout 300s;
+
+    # Buffers used for reading response from a disk
+    # Default: 2 32k
+    output_buffers 2 1m;
+}
+pid ${IMAGEDIR}/nginx.pid;
+daemon on;
+"
```
Comment (ggiguash) [inline test/bin/manage_webserver.sh:53]: This should be deleted.

Comment (vanhalenar) [reply test/bin/manage_webserver.sh]: Done

**Pull request merged**